modules/deploy/pages/deployment-option/self-hosted/kubernetes/k-deployment-overview.adoc

```diff
@@ -5,106 +5,118 @@
+TIP: If you are already using the Redpanda Helm chart and want to migrate to the latest Redpanda Operator for better lifecycle management, see xref:upgrade:migrate/kubernetes/helm-to-operator.adoc[Migrate from the Redpanda Helm chart].
 
 === Helm
 
-https://helm.sh/docs[Helm^] is a package manager for Kubernetes, which simplifies the process of defining, installing, and upgrading Kubernetes applications. Helm uses charts, a collection of files that describe a related set of Kubernetes resources, to deploy applications in a Kubernetes cluster.
+https://helm.sh/docs[Helm^] is a package manager for Kubernetes that simplifies defining, installing, and upgrading Kubernetes applications. Helm uses charts, a collection of files describing Kubernetes resources, to deploy applications in a Kubernetes cluster.
 
-The Redpanda Helm chart generates and applies all the manifest files you need for deploying Redpanda in Kubernetes, including:
+The Redpanda Helm chart provides all the manifest files required to deploy Redpanda in Kubernetes, including:
 
-- The StatefulSet to manage the Redpanda brokers.
-- A Headless ClusterIP Service for internal communication with the Redpanda cluster.
-- A NodePort Service for external communication with the Redpanda cluster.
+- A StatefulSet to manage Redpanda brokers
+- A Headless ClusterIP Service for internal communication with the Redpanda cluster
+- A NodePort Service for external communication with the Redpanda cluster
 
-The Redpanda Helm chart comes with default settings that enable a straightforward deployment out of the box. However, Helm also offers customization by allowing you to override these default values.
 
-You can override the defaults in your own YAML files using the `--values` option or directly in the command line with the `--set` option. These changes in Helm chart values not only influence the configuration of the Redpanda cluster, but also determine what Kubernetes resources are deployed and how these resources are configured.
+[[helm-and-redpanda-operator]]
+=== Redpanda Operator
 
-=== Helm and Redpanda Operator
+The Redpanda Operator provides two deployment modes controlled by the `useFlux` flag. The modes differ in how reconciliation is handled:
 
-The recommended option for deploying Redpanda in Kubernetes is a combination of <<Helm>> and the Redpanda Operator. While the Helm chart facilitates the deployment of Redpanda, the Redpanda Operator manages the lifecycle of Redpanda clusters. The operator uses https://fluxcd.io/flux/concepts/[Flux^] to automate the deployment of the Redpanda Helm chart through the HelmRelease resource.
+- *Flux-managed mode (`useFlux: true`)*:
+When `useFlux` is set to `true` (default), the Redpanda Operator delegates reconciliation to https://fluxcd.io/flux/concepts/[Flux^] controllers. The workflow is as follows:
```

Comment:
I'm not sure how to best phrase this but `useFlux` is an optional boolean. Meaning there are 3 states: true, false, null.

If `useFlux` is explicitly set to `true` or `false` it will be interpreted as that value.

If `useFlux` is `null` either explicitly or not set[^1], it will be interpreted as the value of a CLI flag (Let's call it`--use-flux-default` but the name is TBD) that Rafal is in the process of adding. The flag will default to `true`, meaning that `useFlux: null` will be interpreted as `true` for "standard" installations of the operator.

We can omit documenting the CLI flag for now but I wanted to make sure that you're aware of it.

The weird dance of value interpretation is so we can avoid impacting the experience of existing users at the initial release and later switch the default to `false` followed be the official removal of this flag and flux in its entirety.

Reply:
Thanks, Chris. I'll keep this in mind when we come to doc the new CLI flag.